packages/react/src/components/sideNavigation/mainLevel/MainLevel.tsx

```diff
@@ -0,0 +1,198 @@
+import xor from 'lodash.xor';
+import React, { cloneElement, isValidElement, ReactNode, useContext } from 'react';
+
+import classNames from '../../../utils/classNames';
+import styles from './MainLevel.module.scss';
+import SideNavigationContext from '../SideNavigationContext';
+import { FCWithName } from '../../../common/types';
+import { IconAngleDown, IconLinkExternal } from '../../../icons';
+
+export type MainLevelProps = {
+  /**
+   * If `true`, the item will be marked as active
+   */
+  active?: boolean;
+  /**
+   * Additional class names to apply to the side navigation main level
+   */
+  className?: string;
+  /**
+   * The id of the side navigation main level.
+   */
+  id: string;
+  /**
+   * href attribute of the side navigation main level
+   */
+  href?: string;
+  /**
+   * Optional icon of the side navigation main level.
+   */
+  icon?: React.ReactNode;
+  /**
+   * Index of the side navigation main level.
+   */
+  index?: number;
```

Comment:
It seems this is overwritten by the parent component SideNavigation?

Reply:
I guess this is needed for that overwrite. But it could be mentioned in the property comment.
- [] add comment of index's purpose